### ISSUE TITLE
Containers: Boolean-type Radio fields are not parsed properly on container code

### ISSUE THREAD

**Issue by joseorsini** [dotCMS/core#10869]
If a content type has a Boolean-type radio field, system allows to print directly on the Velocity context the field's value if a content is added to a container.

## Expected Behavior
Regardless of the true|false combination, which may vary depending on the DB, the right field's value should be printed for every single content added to a container. 

## Current Behavior
For boolean-type radio fields, it always show "false" as field's value, even if the content has the field's value set to "true".

The [ContentletServices class](https://github.com/dotCMS/core/blob/master-4.0/dotCMS/src/main/java/com/dotmarketing/services/ContentletServices.java#L234-L250) assigns to the mapped Velocity variable a string-parsed value from the original boolean object that holds the field's value, and during local troubleshooting, the Velocity variable gets its value right on this code.

However, once the Velocity code is parsed, it switches back to "false".

## Known Workarounds
None.

## Steps to Reproduce (for bugs)

1. Create a content type with the following fields:
    - Title: Text, indexed, user-searchable, show on listing.
    - Boolean 1, Radio, indexed, user-serchable, show on listing.
       - true|1.
       - false|0.
    - Boolean 2, Radio, indexed, user-serchable, show on listing.
       - true|1.
       - false|0.
2. Create a Test Container with the following code: 

```
<div class="row columns">
    <h1>$title</h1>
    <p> Before Reset: <br>
        Boolean #1 = $!{boolean1} <br>
        Boolean #2 = $!{boolean2}
    </p>
    #set($boolean1 = false)
    #set($boolean2 = false)
    
    <p> After Reset: <br>
        Boolean #1 = $!{boolean1} <br>
        Boolean #2 = $!{boolean2}
    </p>
</div>
```

3. Create a Test template and parse the Container created on step 2.

4. Add a page and use the Test Template created in step 3.

5. Create 3 different contents under this test content type:

- Content 1:
    - Title: Test 001.
    - Boolean 1: True
    - Boolean 2: True.
- Content 2:
    - Title: Test 002.
    - Boolean 1: False
    - Boolean 2: False.
- Content 3:
    - Title: Test 003.
    - Boolean 1: True
    - Boolean 2: True.

6. On Edit Mode, add these 3 contents to the Test container on the Test page you created.

Screenshot: 

<img width="1103" alt="screen shot 2017-03-02 at 10 07 40 am" src="https://cloud.githubusercontent.com/assets/1717110/23512868/1fdb00ba-ff30-11e6-8ee4-3206728c4abe.png">

As you can see, contents with fields set to true are labeled as false, despite the ContentletServices code referenced above sets all velocity variables with the right value, set as string-parsed value from the original field's value object.

## Context
Reported by customer: 
https://my.dotcms.com/tickets/detail.dot?id=ade3349e-dbf1-4d79-a5d9-964b8a1c9eae

## Your Environment

* dotCMS 3.7.1.
* Google Chrome, Firefox.
* macOS Sierra (App, DB).
* Tomcat 8.0.18.
* Oracle Java JDK 8u60.
* Postgres 9.4.1.

**Comment by cfalzone** [comment]
Not that this behavior can be reproduced with ANY content field, not just booleans.  I also tested Text and Textarea fields and they produce the same behavior

**Comment by oarrietadotcms** [comment]
The bug was introduced by this change: https://github.com/dotCMS/core/issues/10101

With this change ContainerServices.java prints the code for every structure in the container, before it was just printing a call to $containerAPI.getStructureCode()

The client is doing velocity assignments in the container code of one structure and velocity is parsing it before getting the content values, so the assignment takes priority against any content value.

The developer assigned needs to talk to Will about this.

**Comment by wezell** [comment]
We need to figure out how to properly scope and  set/unset these variables in the ContainerServices. without calling the $containerAPI.getStructureCode API on every render which caused a  performance regression.

**Comment by agomez-dotcms** [comment]
Hi Guys,

After analysis and investigation, this issue does not seem introduced by https://github.com/dotCMS/core/issues/10101 but **it was actually originally caused by a couple of lines of code that were left in class ContainerServices after following cherry-pick**: https://github.com/dotCMS/core/commit/0ce83f3d9bf6d6df896aab4a8e32d421a138d692.

To be removed:
```
sb.append("#set ($structureCode" ).append(" = $containerAPI.getStructureCode(\"").append( container.getIdentifier() ).append("\", \"$ContentletStructure\"))" );
```

Additionally, it was discovered that this issue:
- Is present in 3.7.1 and 4.0.1. Hence, **Support team may remove the problematic lines of code for corresponding patch releases**:
3.7.1: https://github.com/dotCMS/core/blob/3.7.1/src/com/dotmarketing/services/ContainerServices.java#L168-L169
4.0.1: https://github.com/dotCMS/core/blob/4.0.1/dotCMS/src/main/java/com/dotmarketing/services/ContainerServices.java#L171-L172
- Is not present in current master-4.1. **Hence, this ticket will be marked as "Done"** as it does not need to be fixed for 4.1.0.
master-4.1: https://github.com/dotCMS/core/blob/master-4.1/dotCMS/src/main/java/com/dotmarketing/services/ContainerServices.java#L169-L173

Thanks,
Anibal
